### PR TITLE
Transform top collections nuxt3 way

### LINKED ISSUE #4364: Follow up top collection data volume and & change v2
// TODO:
- missing translation 
- missing some types `(nft)`
- transform `<TopCollections />` component nuxt3 way
  - #4381 	
- refactor `timeFilters` with `<NeoButton />` from #4330 
<details>
  <summary>Code</summary>
  
```vue
      <b-field>
        <p class="control">
          <NeoButton :class="{ active: state.timeRange == 'Day' }" label="24 Hours" @click="setTimeRange('Day')" />
        </p>
        <p class="control">
          <NeoButton :class="{ active: state.timeRange == 'Day' }" label="7 Days" @click="setTimeRange('Week')" />
        </p>
        <p class="control">
          <NeoButton :class="{ active: state.timeRange == 'Day' }" label="30 Days" @click="setTimeRange('Month')" />
        </p>
        <p class="control">
          <NeoButton :class="{ active: state.timeRange == 'Day' }" label="All" @click="setTimeRange('All')" />
        </p>
      </b-field>
```
  
</details>

_Originally posted by @roiLeo in https://github.com/kodadot/nft-gallery/pull/4332#pullrequestreview-1183865135_

### LINKED ISSUE #4381: transform <TopCollections /> component nuxt3 way
from
- #4364

### LINKED ISSUE #4419: add translation to top collections
translation

## Ref
- #4364

### PR DESCRIPTION
**Pull request by daiagi**
Hi Guys
I'm actually new to Vue so i hope what i did here is sensible
and hope not to waste too much of your time correcting me

The biggest difference is the fetching of collection entities and then their collection sales

1.onmounted calls fetchCollectionsSeries
2. in fetchCollectionsSeries use "useGraphql" to fetch 'topCollectionList' and set result to a ref
3. watcher watching that ref triggers and calls 'fetchCollectionsSales'
4. in fetchCollectionsSeries use "useGraphql" to fetch 'collectionsSales' and set result to a ref
5. another watcher watches that ref and triggers 'proccessData'
6. finally 'proccessData' sets the combined results to 'data' ref and we're done

add typings too

**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Closes #4381
- [x] Closes #4419 
- [x] Closes #4364  
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts - had conflict TopCollections.vue, merged them
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Review by roiLeo — CHANGES_REQUESTED** [review]
I think it would help to take a look at `useCarouselSpotlight()` or useCarousel.ts file and base your code on it.

The goal is to create a `useTopCollection()` method that would fetch data

File: components/landing/TopCollections.vue [hunk 1]
```diff
@@ -106,91 +100,128 @@ import {
   weeklyrangeVolume,
 } from '@/components/series/utils'
 import { TimeRange } from '@/components/series/types'
-import { Collection } from '../rmrk/service/scheme'
+import { Interaction } from '../rmrk/service/scheme'
+import BasicImage from '@/components/shared/view/BasicImage.vue'
+import { Ref, ref } from 'vue'
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/landing/TopCollections.vue]: what is imported by default?

[the remark this answers is not in the document]
Comment (roiLeo) [reply components/landing/TopCollections.vue]: `import { Ref, ref } from 'vue'`

**Comment by daiagi** [comment]
@roiLeo 
alrighty, how is it now?

**Comment by roiLeo** [comment]
Hey! I'm adding @preschian who can help for the review.

Small stuff that I wonder:
- do you need to declare `isLoading`? can't you just use `loading` from useGraphql?
- I would separete the `watch` in two different const (maybe create `useCollectionsSeries` & `useCollectionsSales`)

**Comment by daiagi** [comment]
@roiLeo 
>  do you need to declare isLoading? can't you just use loading from useGraphql?

well, I don't really care for the indivdual loading of topCollections and their sales
I only care if they return with error so i can terminate early
I only want to let the UI know loading is done when proccessData is done and the result is ready to be used

 > I would separete the watch in two different const (maybe create useCollectionsSeries & useCollectionsSales)

Ok, did that, but still is useTopCollections i need to watch twice,
once for the result of useTopCollectionList and the other for the result of useCollectionsSales

File: components/landing/topCollections/useTopCollections.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    events: Interaction[]
+  }[]
+}
+
+type FetchCollectionsSalesResult = {
+  collectionsSales: CollectionSales[]
+}
+//refs
+
+const fetchCollectionsSeriesResult = ref<FetchCollectionsSeriesResult | null>(
```
[the remark this answers is not in the document]
Comment (daiagi) [reply components/landing/topCollections/useTopCollections.ts]: tbh I don't know

why is it better?

File: components/landing/topCollections/utils/useTopCollections.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+export const useTopCollections = (limit: number) => {
+  const topCollectionWithVolumeList = useState<CollectionEntityWithVolumes[]>(
+    'topCollectionWithVolumeList',
+    () => []
+  )
+  const fetchError = ref(null)
+  const isLoading = ref(true)
+  const { data: topCollections, error: fetchTopCollectionsError } =
+    useTopCollectionList(limit)
```
Comment (roiLeo) [inline components/landing/topCollections/utils/useTopCollections.ts:53]: can't you use `loading` from `useTopCollectionList`?

File: components/landing/topCollections/utils/useCollectionsSales.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  collectionsSales: CollectionSales[]
+}
+
+export const useCollectionsSales = (ids: string[]) => {
+  const CollectionSalesList = useState<CollectionSales[] | null>(
+    'CollectionSalesList'
+  )
+
+  const { data, error, loading } = useGraphql({
+    queryPrefix: 'rmrk',
```
Comment (roiLeo) [inline components/landing/topCollections/utils/useCollectionsSales.ts:13]: could we move this query to `subsquid` since it's used on other chains?

File: components/landing/topCollections/utils/useTopCollections.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      if (fetchTopCollectionsErrorValue) {
+        fetchError.value = fetchTopCollectionsErrorValue
+        isLoading.value = false
+        return
+      }
+      if (collectionEntities) {
+        const { data: collectionSales, error: fetchCollectionsSalesError } =
+          useCollectionsSales(collectionEntities.map((c) => c.id))
+
+        watch(
```
Comment (roiLeo) [inline components/landing/topCollections/utils/useTopCollections.ts:66]: nested watch? can't we do otherwise?

File: components/landing/topCollections/utils/useTopCollections.ts (hunk 3 above)
Comment (daiagi) [reply components/landing/topCollections/utils/useTopCollections.ts:53]: if i do that then the loading state will turn to true prematurely.
it will turn to true when useTopCollectionList is done
but i still need to fetch sales and merge the lists together

File: components/landing/topCollections/utils/useCollectionsSales.ts (hunk 4 above)
Comment (daiagi) [reply components/landing/topCollections/utils/useCollectionsSales.ts:13]: sure thing

File: components/landing/topCollections/utils/useTopCollections.ts (hunk 5 above)
Comment (daiagi) [reply components/landing/topCollections/utils/useTopCollections.ts:66]: I thought it was easier to read this way
can move the internal watch outside if you think that's better

Comment (daiagi) [reply components/landing/topCollections/utils/useTopCollections.ts:66]: I think the way to avoid this would be fetch all the necessary data in single query

**Comment by yangwao** [comment]
is this good to go?

**Comment by daiagi** [comment]
> is this good to go? 
> 
> - #4394 was merged, probably merge here?

No. Not yet 
I will rewrite the query to fetch all info in one go
Eliminating the need for nested watch

**Comment by daiagi** [comment]
@roiLeo 
My initial though of uniting the queries is invalid (series type doesn't have nfts field)
What i did instead is to allow useGraphql to accept refs for data, error, loading

this enables to do something like this:

```ts
const data2 = ref(); 
{data: data1} = useGraphql({
    ...
})


watch (data1, () => {
    // do some calcualtion that depnds on data1
    // const info_from_data1 = ...something with data1

   const {data: data2} useGraphql({
        variables: {
            info_from_data1
        },
        data: data2
        ...
    })

})

watch(data2, () => {
    //complete proccessing
})
```

in contrast with the nested watch situation:

```ts
{data: data1} = useGraphql({
    ...
})

watch (data1, () => {
    // do some calcualtion that depnds on data1
    // const info_from_data1 = ...something with data1

   const {data: data2} useGraphql({
        variables: {
            info_from_data1
        },
        ...
    })

    watch(data2, () => {
        //complete proccessing
    })

})


```

is this acceptable?

File: locales/en.json [hunk 6]
```diff
@@ -65,6 +65,14 @@
...
     "landingSubtitle": "Kusama NFTS"
   },
+  "topCollections": {
+    "timeFrames": {
+      "7 Days": "7 Days",
+      "30 Days": "30 Days",
+      "90 Days": "90 Days",
+      "all": "All"
+    }
+  },
```
Comment (roiLeo) [inline locales/en.json:75]: same thing will be done in #4355

**Review by roiLeo — APPROVED** [review]
I'm spending way too much time trying to find another way to do it, for now we can settle for this solution and refactor the code later if needed.

**Comment by daiagi** [comment]
@yangwao
Ready to merge

**Comment by yangwao** [comment]
pay 120 usd 
seems this was a great one heavy lifting!

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $120 @ 30.07 USD/KSM ~ 3.991 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x57a0a7b1d883693139718eed6506b18b1d25463b7dce43c0a26a27fbd7a73eb6](https://kusama.subscan.io/extrinsic/0x57a0a7b1d883693139718eed6506b18b1d25463b7dce43c0a26a27fbd7a73eb6)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)
